### DISCUSSION TITLE
Suppress auto instrumentation trace spans in open telemetry

### DISCUSSION THREAD

**Discussion by vinodkumarpandiyan** [open-telemetry/opentelemetry-java#6815] [Q&A]
Hi Team,

We want to disable auto instrumentation ( java) of trace spans in our application. We have manually created spans for our need. We need only those spans. 

I try to disable it with below configuration. Unfortunately its not working as expected.
otel.instrumentation.common.default-enabled=false

Is there any issues with above configuration? if there is any alternate way, could you advise?

Thanks

**Comment by trask** [comment]
If you don't want any autoinstrumentation, it may be worth considering using the SDK directly

**Reply by vinodkumarpandiyan** [reply]
auto instrumentation library is used in one of the common component. We do not want to remove it in the common component since its being used by other apps. In our application, i want to disable it. 

Is there any way to control that?

**Reply by trask** [reply]
is this what you're looking for? https://opentelemetry.io/docs/zero-code/java/agent/disable/#disabling-the-agent-entirely

**Reply by vinodkumarpandiyan** [reply]
Our need is, we have few manual instrumentation for creating spans. we do not want other spans. 
If we disable the agent entirely, trace will not be created at all right.

**Reply by trask** [reply]
if you don't want to use the SDK to do the exporting (but rather still want to use the Java agent for the exporting), you could try https://opentelemetry.io/docs/zero-code/java/agent/disable/#enable-only-specific-instrumentation

keep in mind: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7201
